File: frontend2/Rugby/src/app/classes/associations/ActiveRoster.ts
import {Game} from '../game';
import {Athlete} from '../athlete';

export class ActiveRoster {
  constructor(
    private id: number,
    public athlete?: Athlete,
    public gameId?: number,
    public number?: number
  ) {
    this.id = id ? id : 0;
    this.athlete = athlete ? athlete : new Athlete();
    this.number = number ? number : 0;
  }
}

File: backend/src/main/java/pt/isel/rugby/model/ActiveRoster.java
package pt.isel.rugby.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "activeRoster")
@Data
public class ActiveRoster implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    private Athlete athlete;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    private Game game;

    @Column
    private Long number;

}
